eth_client/lib/eth_client/contract.ex

```diff
@@ -18,26 +18,79 @@ defmodule EthClient.Contract do
 
   defp parse_abi([], acc), do: {:ok, acc}
 
-  defp parse_abi([%{"type" => "function"} = method_map | tail], acc) do
+  defp parse_abi([%{"type" => "function", "name" => name} = method_map | tail], acc) do
     method = %{
-      name: String.to_atom(method_map["name"]),
+      name: name,
       state_mutability: method_map["stateMutability"],
       inputs: Enum.map(method_map["inputs"], fn input -> input["name"] end),
       # What's the difference between type and internal type?
       input_types: Enum.map(method_map["inputs"], fn input -> input["internalType"] end)
     }
 
-    name_snake_case = String.to_atom(Macro.underscore(method_map["name"]))
+    name_snake_case = String.to_atom(Macro.underscore(name))
     function = build_function(method)
     acc = Map.put(acc, name_snake_case, Code.eval_quoted(function) |> elem(0))
 
     parse_abi(tail, acc)
   end
 
+  defp parse_abi([%{function: name} = method_map | tail], acc) do
+    function = build_function_by_hash(method_map)
+
+    selector_atom = name
+                    |> Macro.underscore()
+                    |> String.to_atom()
+
+    acc = Map.put(acc, selector_atom, Code.eval_quoted(function) |> elem(0))
```

Comment:
as a suggest, i would match `Code.eval_quoted(function) |> elem(0))` with a variable to improve readability. It would look like: 
'
{term, _bind} = Code.eval_quoted(function)
acc = Map.put(acc, selector_atom, term)
`